src/transformers/modeling_utils.py

```diff
@@ -1882,6 +1882,32 @@ def gradient_checkpointing_enable(self, gradient_checkpointing_kwargs=None):
             # the gradients to make sure the gradient flows.
             self.enable_input_require_grads()
 
+    def _set_gradient_checkpointing(self, enable: bool = True, gradient_checkpointing_func: Callable = None):
+        gradient_checkpointing_set = False
+
+        # We default it to `torch.utils.checkpoint.checkpoint`
+        if gradient_checkpointing_func is None:
+            gradient_checkpointing_func = torch.utils.checkpoint.checkpoint
+
+        # Apply it on the top-level module in case the top-level modules supports it
+        # for example, LongT5Stack inherits from `PreTrainedModel`.
+        if hasattr(self, "gradient_checkpointing"):
+            self._gradient_checkpointing_func = gradient_checkpointing_func
```

Comment:
```suggestion
            self._checkpoint = gradient_checkpointing_func
```
no what would appear best for users to know that its basically just torch.utils.checkpoint

Reply:
Hmmm what I like with `_gradient_checkpointing_func` is that it tells users that it is a function + 'checkpoint' seems a bit ambiguous to me (it can sound like a model checkpoint?)